client/header/activity-panel/index.js

```diff
@@ -190,40 +213,52 @@ export class ActivityPanel extends Component {
 	}
 
 	getPanelContent( tab ) {
+		const { hasUnreadOrders, query, hasUnapprovedReviews } = this.props;
+		const { task } = query;
+
 		switch ( tab ) {
 			case 'inbox':
 				return <InboxPanel />;
 			case 'orders':
-				const { hasUnreadOrders } = this.props;
 				return <OrdersPanel hasActionableOrders={ hasUnreadOrders } />;
 			case 'stock':
 				return <StockPanel />;
 			case 'reviews':
-				const { hasUnapprovedReviews } = this.props;
 				return (
 					<ReviewsPanel
 						hasUnapprovedReviews={ hasUnapprovedReviews }
 					/>
 				);
 			case 'help':
-				const { query } = this.props;
-				const { task } = query;
 				return <HelpPanel taskName={ task } />;
 			default:
 				return null;
 		}
 	}
 
 	renderPanel() {
+		const { updateOptions } = this.props;
 		const { isPanelOpen, currentTab, isPanelSwitching } = this.state;
-
 		const tab = find( this.getTabs(), { name: currentTab } );
+
 		if ( ! tab ) {
 			return (
 				<div className="woocommerce-layout__activity-panel-wrapper" />
 			);
 		}
 
+		const clearPanel = () => {
+			this.clearPanel();
+		};
+
+		if ( currentTab === 'setup' ) {
+			// Ensure that if the user is trying to get to the task list they can see it even if
+			// it was dismissed.
+			updateOptions( { woocommerce_task_list_hidden: 'no' } );
+			getHistory().push( getAdminLink( 'admin.php?page=wc-admin' ) );
```

Comment:
The url comes out to this:
```
http://tangaroa.test/wp-admin/http://tangaroa.test/wp-admin/http://tangaroa.test/wp-admin/admin.php?page=wc-admin
```
`getAdminLink` is for external links. Sorry about this not being documented anywhere! This will look more like

https://github.com/woocommerce/woocommerce-admin/blob/d2be77c0176773bad95ca720d195b3cfad9fdea4/client/task-list/tasks.js#L106

Reply:
Whoops! Thanks, I'll fix that